public/app/core/components/NavBar/Next/NavBarMenu.tsx

```diff
@@ -0,0 +1,272 @@
+            <ul className={styles.itemList}>
+              {navItems.map((link) =>
+                linkHasChildren(link) ? (
+                  <MenuItem key={link.text} link={link} isActive={isMatchOrChildMatch(link, activeItem)}>
+                    <ul>
+                      {link.children.map(
+                        (childLink) =>
+                          !childLink.divider && (
+                            <NavBarMenuItem
+                              key={`${link.text}-${childLink.text}`}
+                              isActive={activeItem === childLink}
+                              isDivider={childLink.divider}
+                              onClick={() => {
+                                childLink.onClick?.();
+                                onClose();
+                              }}
+                              styleOverrides={styles.item}
+                              target={childLink.target}
+                              text={childLink.text}
+                              url={childLink.url}
+                              isMobile={true}
+                            />
+                          )
+                      )}
+                    </ul>
+                  </MenuItem>
+                ) : link.id === 'saved-items' ? (
+                  <MenuItem
+                    key={link.text}
+                    link={link}
+                    isActive={isMatchOrChildMatch(link, activeItem)}
+                    className={styles.savedItems}
+                  >
+                    <em className={styles.savedItemsText}>No saved items</em>
+                  </MenuItem>
+                ) : (
+                  <li key={link.text} className={styles.flex}>
+                    <NavBarItemWithoutMenu
+                      className={styles.itemWithoutMenu}
+                      elClassName={styles.fullWidth}
+                      label={link.text}
+                      url={link.url}
+                      target={link.target}
+                      onClick={() => {
+                        link.onClick?.();
+                        onClose();
+                      }}
+                      isActive={link === activeItem}
+                    >
+                      <div className={styles.savedItemsMenuItemWrapper}>
+                        {link.img && (
+                          <img
+                            src={link.img}
+                            alt={`${link.text} logo`}
+                            height="24"
+                            width="24"
+                            style={{ borderRadius: '50%' }}
+                          />
+                        )}
+                        {link.icon && <Icon name={link.icon as IconName} size="xl" />}
+                        <span className={styles.linkText}>{link.text}</span>
+                      </div>
+                    </NavBarItemWithoutMenu>
+                  </li>
+                )
+              )}
```

Comment:
I think there's another way we could express all of this that would make it a bit easier to read, rather than the very big nested ternaries 😇 

We could have a `<RenderNavItem />` component that is something like

```
const RenderNavItem = () => {
  if (linkHasChildren(link)) {
    return ...
  } else if (...) {
    return ...
  } else {
    return ...
  }
```

Could break each out into separate components (in the same file?) to make it a bit easier to read also?